### PR TITLE
telegram-desktop: update to git master to fix build with Qt 6.8

### PR DESCRIPTION
**Pull request by K900**
Hopefully we can get a new release before staging-next is done, but this is probably an OK fallback.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/applications/networking/instant-messengers/telegram/telegram-desktop/qt68.patch [hunk 1]
```diff
@@ -0,0 +1,355 @@
...
+ 	};
+diff --git a/Telegram/SourceFiles/lang/lang_cloud_manager.cpp b/Telegram/SourceFiles/lang/lang_cloud_manager.cpp
+index 8c98b96d1..d7692e0be 100644
+--- a/Telegram/SourceFiles/lang/lang_cloud_manager.cpp
++++ b/Telegram/SourceFiles/lang/lang_cloud_manager.cpp
+@@ -434,7 +434,6 @@ void CloudManager::requestLanguageAndSwitch(
+ 
+ void CloudManager::sendSwitchingToLanguageRequest() {
+ 	if (!_api) {
+-		_switchingToLanguageId = -1;
```
Comment (NickCao) [inline pkgs/applications/networking/instant-messengers/telegram/telegram-desktop/qt68.patch:51]: The `QChar` constructs are self-explanatory, but what's with this line?

Comment (K900) [reply pkgs/applications/networking/instant-messengers/telegram/telegram-desktop/qt68.patch:51]: This is a QString, which used to be initialized via  implicit QString(QChar(int)). I have no idea what the intent was, but thankfully the value isn't actually ever used.

**Review by NickCao — COMMENTED** [review]
Have been using 5.6.1 for a while, should be good as long as this builds.

**Comment by K900** [comment]
I've merged the 5.6.1 update to master, so this should just have the Qt 6.8 patch after the automerge runs.

**Pull request merged**